src/xml_parser_filelists.c

```diff
@@ -195,6 +210,9 @@ cr_start_handler(void *pdata, const xmlChar *element, const xmlChar **attr)
                 cr_xml_parser_warning(pd, CR_XML_WARNING_UNKNOWNVAL,
                                       "Unknown file type \"%s\"", val);
         }
+
+        if (!pd->last_digest)
+            pd->last_digest = cr_find_attr("hash", attr);
```

Comment:
I think you have to make a copy here, the `attr` don't live beyond the `cr_start_handler`.

Reply:
> What should I be doing?

Ok, I found a similar case with `cpid` field. I will use the same approach/

Reply:
Done. That was not the coredump : (

Reply:
If I am not mistaken you can add it right in to the `pd->pkg->chunk` and then in the `cr_end_handler` just copy the pointer.
You also should reset the `pd->last_digest` or don't check here if it is already set otherwise you will set it only once and all the files will have the same digest.